frank/utilities.py

```diff
@@ -115,6 +120,45 @@ def __init__(self, uv, V, weights, bin_width):
         self._uv_left = bins[:-1][idx]
         self._uv_right = bins[1:][idx]
 
+        # Create a mapping from the complete set of bins to the masked ones
+        bin_number = np.cumsum(idx)-1
+        bin_number[~idx] = -1
+        self._bin_number = bin_number
+
+    def determine_uv_bin(self, uv):
+        r"""Determine the bin that the given uv points belong too.
+        
+        Parameters
+        ----------
+        uv : array, unit = :math:`\lambda`
+            Baselines to determine the bins of
+        
+        Returns
+        -------
+        idx : array,
+            Bins that the uv point belongs to. Will be -1 if the bin does not
```

Comment:
I don't quite follow the rationale for using `-1`. If any of the uv points really do belong to the last bin, won't the return be degenerate with that for points that are at a baseline beyond the last bin? Would it be more intuitive to just return `float('inf')`?

Should there also be a special case (unique return) for points that are at a smaller baseline than the first bin, say `float('-inf')`?

Reply:
Idx is an array of int, not float, so something is needed to specify a bad value. -1 is fine for a bad value, but I'm happy with other choices.

Reply:
I thought `float(`inf`) behaves as an integer, e.g., https://stackoverflow.com/questions/40445920/represent-infinity-as-an-integer-in-python-2-7.

If so, I think this would be good to have `float(`-inf`) and `float(`inf`) for baselines smaller (larger) than the left edge of the first bin (right edge of the last bin).

Reply:
What about ones in between?

Reply:
From the link, "You can't represent infinity as an integer, but the floating point representation has a special case to allow it."

Sorry what do you mean by "ones in between"?

Reply:
> From the link, "You can't represent infinity as an integer, but the floating point representation has a special case to allow it."

Yes indeed, but we are using integers not floating points for index. floating point indices do not make sense and are forbidden in python.

There are empty bins that have been removed from the UVDataBinner

Reply:
> > From the link, "You can't represent infinity as an integer, but the floating point representation has a special case to allow it."
> 
> Yes indeed, but we are using integers not floating points for index. floating point indices do not make sense and are forbidden in python.

Right, that's kind of the point of using `float(inf)` or more generally something that isn't an integer at which `uv` can't be indexed. I thought it confusing to return `-1` for cases where the baseline doesn't belong to any bin.

> 
> There are empty bins that have been removed from the UVDataBinner

Ok that's what I thought, but then any points "in between" have to belong to a bin, because they've populated it, so that baseline range is no longer empty/removed, right?

Reply:
> Right, that's kind of the point of using float(inf) or more generally something that isn't an integer at which uv can't be indexed. I thought it confusing to return -1 for cases where the baseline doesn't belong to any bin.

But it _has_ to be an int because I'm returning an array of indices.

> Ok that's what I thought, but then any points "in between" have to belong to a bin, because they've populated it, so that baseline range is no longer empty/removed, right?

The only time you'll get points in empty bins, before the first, or after the last is if the uv set passed in differs from the set used to create the UVDataBinner. In that case all three are possible. If the datasets are the same, then none of them are possible.